sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/checks/MobileDatabaseEncryptionKeysCheck.kt

```diff
@@ -0,0 +1,94 @@
+package org.sonarsource.kotlin.checks
+
+import org.jetbrains.kotlin.com.intellij.psi.PsiElement
+import org.jetbrains.kotlin.psi.KtCallExpression
+import org.jetbrains.kotlin.psi.KtDotQualifiedExpression
+import org.jetbrains.kotlin.psi.KtExpression
+import org.jetbrains.kotlin.psi.KtNamedFunction
+import org.jetbrains.kotlin.psi.KtReturnExpression
+import org.jetbrains.kotlin.psi.KtStringTemplateExpression
+import org.jetbrains.kotlin.psi.psiUtil.anyDescendantOfType
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.resolve.DescriptorToSourceUtils
+import org.jetbrains.kotlin.resolve.calls.callUtil.getResolvedCall
+import org.jetbrains.kotlin.resolve.calls.model.ExpressionValueArgument
+import org.jetbrains.kotlin.resolve.calls.model.ResolvedCall
+import org.sonar.check.Rule
+import org.sonarsource.kotlin.api.CallAbstractCheck
+import org.sonarsource.kotlin.api.ConstructorMatcher
+import org.sonarsource.kotlin.api.FunMatcher
+import org.sonarsource.kotlin.api.SecondaryLocation
+import org.sonarsource.kotlin.api.predictRuntimeValueExpression
+import org.sonarsource.kotlin.converter.KotlinTextRanges.textRange
+import org.sonarsource.kotlin.plugin.KotlinFileContext
+
+private val CREATE_CHAR_BYTE_ARRAY = FunMatcher(qualifier = "kotlin") { withNames("byteArrayOf", "charArrayOf") }
+private const val SQLITE = "net.sqlcipher.database.SQLiteDatabase"
+private const val ENCRYPTION_KEY = "encryptionKey"
+private const val CHANGE_PASSWORD = "changePassword"
+
+@Rule(key = "S6301")
+class MobileDatabaseEncryptionKeysCheck : CallAbstractCheck() {
+    override fun functionsToVisit() = listOf(
+        ConstructorMatcher(typeName = SQLITE),
+        FunMatcher(qualifier = SQLITE) {
+            withNames(CHANGE_PASSWORD, "openDatabase", "openOrCreateDatabase", "create")
+        },
+        FunMatcher(qualifier = "io.realm.RealmConfiguration.Builder", name = ENCRYPTION_KEY)
+    )
```

Comment:
We call `functionsToVisit()` a lot, every time we visit a call expression. So every time we need to build the list containing the newly constructed matchers. WDYT of making the list static and returning the same instance every time?

Reply:
How about making this as property instead of a function in the CallAbstractCheck

Reply:
I think that's a good idea :)